modules/core/src/main/scala/sangria/execution/ValueCoercionHelper.scala

```diff
@@ -580,98 +582,108 @@ class ValueCoercionHelper[Ctx](
     nodeLocation.toList ++ firstValue.toList
   }
 
-  def isValidValue[In](tpe: InputType[_], input: Option[In])(implicit
-      um: InputUnmarshaller[In]): Vector[Violation] = (tpe, input) match {
-    case (OptionInputType(ofType), Some(value)) if um.isDefined(value) =>
-      isValidValue(ofType, Some(value))
-    case (OptionInputType(_), _) => Vector.empty
-    case (_, None) => Vector(NotNullValueIsNullViolation(sourceMapper, Nil))
-
-    case (ListInputType(ofType), Some(values)) if um.isListNode(values) =>
-      um.getListValue(values)
-        .toVector
-        .flatMap(v =>
+  def isValidValue[In](tpe: InputType[_], input: Option[In], errors: ListBuffer[Violation])(implicit
+      um: InputUnmarshaller[In]): Vector[Violation] = {
+    // TODO: NEED TO ADD VALUES TO `errors` to make it work properly
```

Comment:
This code doesn't work correctly because I don't mutate errors buffer, but this is one of the options how it can be implemented